include/swift/Option/FrontendOptions.td

```diff
@@ -277,6 +277,11 @@ def Raccess_note_EQ : Joined<["-"], "Raccess-note=">,
 def block_list_file
   : Separate<["-"], "blocklist-file">, MetaVarName<"<path>">,
     HelpText<"The path to a blocklist configuration file">;
+
+def wasm_plugin_server_path : Separate<["-"], "wasm-plugin-server-path">,
```

Comment:
Can we merge wasm plugin server into the main plugin server? Having different frontend options for each mode complicates build systems, so it would be nice to unify the plugin servers into single one and switch underlying plugin mechanisms based on plugin extension or file magic.

Reply:
I have a few concerns about bringing wasm into the main plugin server, curious to hear what you folks think:

1. Security: merging the servers will mean that enabling JIT for Wasm would allow library-based plugins to themselves use JIT. This increases risk exposure a bit, though it might be mitigated by AMFI library validation if that's indeed applied to the plugin server.
2. Wasm plugins are currently invoked via `-load-plugin-executable Lib.wasm#Module`, with the server being an impl detail. Meanwhile bona fide server plugins are invoked with `-external-plugin-path Lib#Exec`, and notably A) this doesn't allow supplying an external module name, B) the client would have to locate the executable themselves, as afaik the Driver only does the locating for a pre-existing set of modules. Perhaps the solution to this would be to A) add a `-load-external-plugin-with-module Lib#Exec#Module` to the driver and B) Let the driver convert `-load-plugin-executable Lib.wasm#Module` to `-load-external-plugin-with-module Lib.wasm#<swift-plugin-server>#Module`? Would be nice to have the `-load-external-plugin-with-module` god-option anyway.

Reply:
> 1. Security

Yes, once we will conclude that we cannot satisfy performance expectations without JIT, we need a separate process to minimize such risks, but let's start in a simpler way

> 2. God option

`-load-plugin <library path>:<server path>#<module name>,...` in the current tree seems good to me